x-pack/plugins/security_solution/public/common/lib/cell_actions/default_cell_actions.tsx

```diff
@@ -237,7 +186,128 @@ export const cellActions: TGridCellAction[] = [
         </>
       );
     },
+  ({
+    browserFields,
+    data,
+    ecsData,
+    header,
+    timelineId,
+    pageSize,
+  }: {
+    browserFields: BrowserFields;
+    data: TimelineNonEcsData[][];
+    ecsData: Ecs[];
+    header?: ColumnHeaderOptions;
+    timelineId: string;
+    pageSize: number;
+  }) => {
+    if (header !== undefined) {
+      return ({
+        rowIndex,
+        columnId,
+        Component,
+        closePopover,
+      }: EuiDataGridColumnCellActionProps) => {
+        /* eslint-disable @typescript-eslint/no-shadow */
```

Comment:
The refactoring of the business logic that originally existed in the `const` definition improves the readability 🙏 
It appears that relocating the refactored code from the `const` outside the function to its new home here shadows a few variables. Consider extracting this improved code into a helper function outside this component definition to eliminate the shadowing.